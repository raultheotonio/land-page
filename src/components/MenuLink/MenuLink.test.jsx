import { screen } from '@testing-library/react';
import { MenuLink } from '.';
import renderTheme from '../../utils/RenderTheme';

describe('<MenuLink />', () => {
  it('should render a link', () => {
    renderTheme(
      <MenuLink link="http://localhost" newTab={false}>
        children
      </MenuLink>
    );

    const link = screen.getByRole('link', { name: 'children' });

    expect(link).toBeInTheDocument();
  });

  it('should render a link with a target _self', () => {
    renderTheme(
      <MenuLink link="http://localhost" newTab={false}>
        children
      </MenuLink>
    );

    const link = screen.getByRole('link', { name: 'children' });

    expect(link).toHaveAttribute('target', '_self');
  });

  it('should render a link with a target _blank', () => {
    renderTheme(
      <MenuLink link="http://localhost" newTab>
        children
      </MenuLink>
    );

    const link = screen.getByRole('link', { name: 'children' });

    expect(link).toHaveAttribute('target', '_blank');
  });

  it('should render snapShot', () => {
    renderTheme(
      <MenuLink link="http://localhost" newTab>
        children
      </MenuLink>
    );

    const link = screen.getByRole('link', { name: 'children' });

    expect(link).toMatchSnapshot();
  });
});
