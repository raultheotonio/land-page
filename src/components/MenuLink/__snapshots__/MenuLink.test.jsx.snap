// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuLink /> should render snapShot 1`] = `
<a
  class="sc-bcXHqe eNFlyR"
  href="http://localhost"
  target="_blank"
>
  children
</a>
`;
